index.js

```diff
@@ -101,12 +112,37 @@ class Client {
       this.accessToken = tokens.access
       this.refreshToken = tokens.refresh
 
-      uuid = await requester.do(options, this.accessToken, this.apiUrl)
+      requestResponse = await requester.do(options, this.accessToken, this.apiUrl)
+    }
+
+    const uuid = requestResponse.uuid
+    let timeout = options.timeout
+
+    // FIXME: this might not be optimal. The test should be negated
+    // and then do the *only* subset of poller that needs to be done,
+    // given that we know this is finished. Instead, I think we'll
+    // make one more additional request of status only to find out
+    // that it is still finished.
+    if (requestResponse.status !== 'Finished') {
+      // Compute the initial delay as the larger of the default value
+      // and what is passed in.
+
+      let initialDelay = defaultInitialDelay
+      if (!isNaN(options.initialDelay) && options.initialDelay > initialDelay) {
+        initialDelay = options.initialDelay
+      }
+
+      // However, if the *total* timeout is small, we can use 90% of that
+      // instead, since we'll do just one or two polls anyway.
+
+      initialDelay = Math.min(initialDelay, 0.9 * timeout)
```

Comment:
I'd say, no need for this. Even if caller ends up using `initialDelay > timeout`, then just wait for `timeout` and bail out. This way we keep the promise behind each argument in this corner case:
`timeout` specifies that the function will exit at most that time after the start,
`initialDelay` promises that no actuall API calls are done by function before this time from the start.

Reply:
After thinking about this a bit, I think you are correct.

To make it clear, here is what I had been thinking. In an IDE what you want is a short timeout say 1-5 seconds. That of course is too small to do an analysis. But that's okay, because the point was just to queue something up when there was indeed a change. 

In other words, setting timeout to say 1 second really means:  If the request was cached, return it. If not just submit it and return.